lib/matplotlib/backends/backend_qt5.py

```diff
@@ -460,6 +460,15 @@ def draw_idle(self):
             self._draw_pending = True
             QtCore.QTimer.singleShot(0, self._draw_idle)
 
+    def blit(self, bbox=None):
+        # docstring inherited
+        if bbox is None and self.figure:
+            bbox = self.figure.bbox  # Blit the entire canvas if bbox is None.
+        # repaint uses logical pixels, not physical pixels like the renderer.
+        l, b, w, h = [pt / self._dpi_ratio for pt in bbox.bounds]
+        t = b + h
+        self.repaint(l, self.rect().height() - t, w, h)
```

Comment:
OK, I can't find it - does qt5agg have a `rect` method?  an is `self.rect().height()` equal to `self.renderer.height / self._dpi_ratio`?

Reply:
Yes, that comes to us from qt directly: https://doc.qt.io/qt-5/qwidget.html#rect-prop.